src/protocols/routing/mod.rs

```diff
@@ -147,6 +151,7 @@ fn build_forward_message(
         serde_json::from_str(forwarded_msg)
             .kind(ErrorKind::Malformed, "Unable deserialize forwarded message")?,
     )
+    .id(generate_attachment_id())
```

Comment:
Could you use the function above `generate_message_id`, maybe we could rename it? I do not like having two functions which have exactly the same body.